File: python/burkut/resources/viop.py
"""
VIOP (Futures and Options) Resource
"""

from typing import Any, Dict, List
from burkut.resources.base import BaseResource


class ViopResource(BaseResource):
    """Vadeli İşlem ve Opsiyon Piyasası (VİOP) sözleşmeleri."""

    def list(self) -> List[Dict[str, Any]]:
        """
        Tüm aktif VİOP sözleşmelerini listeler.

        Returns:
            VİOP sözleşmeleri listesi.
        """
        res = self._get("/viop")
        return res if isinstance(res, list) else []

    def get(self, symbol: str) -> Dict[str, Any]:
        """
        Belirli bir VİOP sözleşmesinin detaylarını getirir.

        Args:
            symbol: VİOP sözleşme kodu (Örn: 'F_XU0301026')
        """
        if not symbol or not isinstance(symbol, str):
            raise ValueError("symbol parametresi boş olamaz.")
        return self._get(f"/viop/{symbol.strip().upper()}")

File: nodejs/src/resources/viop.ts
import { BaseResource } from './base';
import type { ViopItem } from '../types';

export class ViopResource extends BaseResource {
  /**
   * Tüm aktif VİOP sözleşmelerini listeler.
   */
  async list(): Promise<ViopItem[]> {
    return this._get<ViopItem[]>('/viop');
  }

  /**
   * Belirli bir VİOP sözleşmesinin detaylarını getirir.
   * @param symbol VİOP sözleşme kodu (Örn: 'F_XU0301026')
   */
  async get(symbol: string): Promise<ViopItem> {
    if (!symbol || typeof symbol !== 'string') {
      throw new Error('symbol parametresi geçerli bir metin olmalıdır.');
    }
    return this._get<ViopItem>(`/viop/${symbol.trim().toUpperCase()}`);
  }
}
